Automattic/sensei
# Add warning on empty quiz response

Fixes #1718 

Adds a new option to quizzes. When this option is enabled, users taking the quiz will see a warning if they have left any responses blank. They may still opt to submit the form with the blank responses.

## Testing

- Create a lesson with a quiz.
- Check the box "Warn user on empty response".
- Take the quiz. Leave some questions blank and ensure that the alert appears.
- Ensure that "Ok" and "Cancel" in the alert behave as expected.
- Test with all different question types (multi-line and file upload were particularly tricky, please test those on various browsers).
- Turn the option off. Ensure that you can submit empty responses without seeing the alert.

## Notes

- This iteration will only warn the user. There is no option to actually prevent the user from submitting if there are empty responses.

- We're just using a JS dialog here. I'm not sure that's what we want from a design perspective. Are there any modal dialogs that we use elsewhere on the frontend in Sensei? Alternatively I could just display a message right in the html. But I think the alert works for now.

- I had to do a couple of hacky things in JS (no surprise there...) but I tried to make the code as clean as possible. Please feel free to call me out on anything nasty, and I can clean it up.

- There are no tests for the JS functionality. @pgk I couldn't find any obvious existing JS tests. I'm assuming we don't have anything set up yet?

- I set up gulp to do more JS minification, and updated the minified files across the board

- The function `sensei_quiz_action_on_empty_response` is set up so that we could add other actions later (e.g. `error`)
